pipeline-modules/app-service/aws/docker-build-n-push.yaml

```diff
@@ -47,9 +52,11 @@ template: |
         - echo Logging in to docker
         - echo $docker_password | docker login -u $docker_username --password-stdin $docker_registry
         - echo Build started on `date`
-        - echo Building the Docker image...
+        {% if use_github_tag == true %}
+        - sha_tag=$$(git log --format="%H" -n 1);
+        - docker build -t {{container_repository}}:$$sha_tag {% for arg in docker_build_args %} {{ arg }} {% endfor %} .
```

Comment:
@pranaySinghDev you don't need to do build twice to achieve this. You can just use docker tag to add the sha_tag after you do the docker build on container_tag. you will still have to push twice but you can get away with one build.

Reply:
One other Q - I'm not sure I understand why the sha wouldn't come thru as the container_tag?

Reply:
we should deprecate this module in favour of separate build and push modules. since we have multiple ways to build now, it doesn’t make sense to have a combined module as it leads to redundant code.
the existing docker-push module already does this.
@dshelley66 that is a good question the problem is, we have place for only one tag in the artefact object but in reality the users might want to push the same image with multiple tags. like `latest`, `<commit sha>`. so we get `latest` from container artifact object and commit sha from git artifact object.
another use-case is that, if user wants to have dev, staging, prod as long live branches, they can override container artifact and create three pipelines that triggers on respective branch push and creates these tag 
- push to dev -> docker tag `dev-latest`, `<commit sha>` -> deploy <commit sha>
- push to staging -> docker tag `staging-latest`, `<commit sha>` -> deploy <commit sha>
- push to prod -> docker tag `prod-latest`, `<commit sha>` -> deploy <commit sha>

so right now we are combining information from container artifact and git artifact to generate final container images. in future we can provide more customisations to user around this to say also include semantic version from git tags, etc.